util/awk.cwl

```diff
@@ -0,0 +1,60 @@
+#!/usr/bin/env cwl-runner
+cwlVersion: v1.0
+class: CommandLineTool
+label: awk
+doc: pattern scanning and processing with awk
+$namespaces:
+  s: http://schema.org/
+
+requirements:
+  DockerRequirement:
+    dockerPull: busybox:latest
+  InlineJavascriptRequirement: {}
+
+inputs:
+  field_separator:
+    doc: string that separaters fields on a line
+    type: string?
+    inputBinding:
+      prefix: -F
+  inherit_format:
+    doc: copy format from target_files input to result output
+    type: boolean?
+  program:
+    doc: AWK program
+    type: string
+    inputBinding:
+      prefix: -e
+  target_files:
+    doc: files to apply program to
+    type: File[]
+    inputBinding:
+      position: 100
+    streamable: true
+  variable_setting:
+    doc: one or more strings of the format VAR=VALUE to set VAR to VALUE
+    type: string[]?
+    inputBinding:
+      prefix: -v
+
+outputs:
+  result:
+    type: stdout
+    format: |-
+      $(inputs.inherit_format && inputs.target_files[0].format ? inputs.target_files[0].format : null)
+    streamable: true
+stdout: awk_result$(inputs.target_files[0].nameext)
```

Comment:
Is the extension really needed?

Reply:
Since extensions often encode meaning it is useful to include it. Perhaps more so when inheriting format. Perhaps I should make it optional depending on the setting of the inherit_format flag.

Reply:
Sure, that works for me!